management/src/main/java/io/micronaut/management/endpoint/health/HealthEndpoint.java

```diff
@@ -79,8 +90,39 @@ public HealthEndpoint(HealthAggregator<HealthResult> healthAggregator,
     @Read
     public Single<HealthResult> getHealth(@Nullable Principal principal) {
         HealthLevelOfDetail detail = levelOfDetail(principal);
+        // due to backward comaptibility all HealthIndicators that are not qualified with Readiness
+        // e.g. are Liveness or without qualifier are returned by this resource
         return Single.fromPublisher(
-                healthAggregator.aggregate(healthIndicators, detail)
+                healthAggregator.aggregate(
+                        Arrays.stream(healthIndicators).
+                                filter(healthIndicator ->
+                                        !Arrays.asList(readinessHealthIndicators).contains(healthIndicator))
+                                .toArray(HealthIndicator[]::new),
+                        detail)
+        );
+    }
+
+    /**
+     * @param principal Authenticated user
+     * @param selector HealthEndpointSelector
+     * @return The health information as a {@link Single}
+     */
+    @Read
+    public Single<HealthResult> getHealth(@Nullable Principal principal, @Selector HealthEndpointSelector selector) {
```

Comment:
Not sure whether using enum in selector isn't too much defensive

Reply:
I think an enum is fine.